public/components/event_analytics/explorer/save_panel/save_panel.tsx

```diff
@@ -50,59 +42,36 @@ export const SavePanel = ({
   savedObjects,
   savePanelName,
   showOptionList,
-  curVisId,
-  spanValue,
   setSubType,
-  metricMeasure,
-  setMetricMeasure,
-  setMetricLabel,
-  metricChecked,
+  isSaveAsMetricEnabled,
 }: ISavedPanelProps) => {
   const [options, setOptions] = useState([]);
   const [checked, setChecked] = useState(false);
-  const [measure, setMeasure] = useState('');
-  const [label, setLabel] = useState([]);
+  const [svpnlError, setSvpnlError] = useState(null);
 
-  const getCustomPabnelList = async (savedObjects: SavedObjects) => {
-    const optionRes = await savedObjects
+  const getCustomPabnelList = async (svobj: SavedObjects) => {
+    const optionRes = await svobj
       .fetchCustomPanels()
       .then((res: any) => {
         return res;
       })
-      .catch((error: any) => console.error(error));
+      .catch((error: any) => setSvpnlError(error));
     setOptions(optionRes?.panels || []);
   };
 
   useEffect(() => {
     getCustomPabnelList(savedObjects);
-  }, []);
+  });
 
   const onToggleChange = (e: { target: { checked: React.SetStateAction<boolean> } }) => {
     setChecked(e.target.checked);
     if (e.target.checked) {
-      setSubType("metric")
+      setSubType('metric');
     } else {
-      setSubType("visualization")
+      setSubType('visualization');
     }
   };
 
-  const onMeasureChange = (selectedMeasures: string) => {
-    setMeasure(selectedMeasures);
-    setMetricMeasure(selectedMeasures);
-  };
-
-  const onLabelChange = (selectedLabels: React.SetStateAction<never[]>) => {
-    setLabel(selectedLabels);
-    setMetricLabel(selectedLabels);
-  };
-
-  useEffect(() => {
-    if (metricChecked) {
-      setChecked(true);
-      setMeasure(metricMeasure);
-    }
-  }, [])
```

Comment:
is this what was causing snapshot change? Is this deletion intended?

Reply:
yes intended, I recall this should also relate to unused variable issues.